src/components/PublishCarForm.component.tsx

```diff
@@ -0,0 +1,306 @@
+import {
+  ButtonOnHoverOppacity,
+  CSSFlexCol,
+  FlexRow,
+  Form,
+  H2,
+  H3,
+  InputStyle,
+  P,
+  ValidationText,
+} from "./styled";
+import Loader from "./styled/Loader.component";
+
+interface FormValues {
+  brand_id: number;
+  model_id?: number;
+  year?: number;
+  state_id?: number;
+  city_id?: number;
+  vin?: string;
+  color_id?: number;
+  odometer: number;
+  condition: string;
+  sale_date: string;
+  price: number;
+  title?: string;
+}
+const VALIDATION_MESSAGES = {
+  BRAND: "Select your car's Brand",
+  MODEL: "Select your car's Model",
+  YEAR: "Enter your car's fabrication year",
+  CITY: "Select your car's city of procedence",
+  VIN: "Enter your Vehicle's Identification Number",
+  COLOR: "Select your car's color",
+  CONDITION: "Pick your car's condition",
+  SALE_DATE: "Select when you want to sell your car",
+  PRICE: "Enter a price for your car",
+};
+export const PublishCarForm = () => {
+  const { data: formsData, error: dataError, loading } = useFormDataQuery();
+  const [
+    fetchModels,
+    { data: modelsData, error: modelsError, loading: modelsLoading },
+  ] = useModelsLazyQuery();
+
+  const [createCar, { data: mutationReturn, loading: mutationLoading }] =
+    useCreateCarMutation();
+  const {
+    register,
+    reset,
+    handleSubmit,
+    watch,
+    setValue,
+    getValues,
+    formState: { errors },
+  } = useForm<FormValues>();
+
+  const threeMonthsAhead = new Date();
+  threeMonthsAhead.setDate(threeMonthsAhead.getDate() + 90);
+
+  const brandId = watch("brand_id");
+  const odometer = watch("odometer");
+  const cityId = watch("city_id");
+
+  const onSubmit: SubmitHandler<any> = async (data) => {
+    const brand = formsData?.brands.find(
+      (brand) => brand.id == getValues("brand_id")
+    )?.name;
+    const model = modelsData?.models.find(
+      (model) => model.id == getValues("model_id")
+    )?.name;
+    data.title = `${brand} ${model} ${getValues("year")}`;
+    for (const key in data)
+      if (key.includes("id")) data[key] = parseInt(data[key]);
+
+    try {
+      await createCar({
+        variables: { object: data },
+        optimisticResponse: {
+          insert_cars_one: data,
+          __typename: "mutation_root",
+        },
+        update(cache, { data }) {
+          cache.modify({
+            fields: {
+              cars(existingCars = []) {
+                const newCar = data?.insert_cars_one;
+                return existingCars.concat(newCar);
+              },
+            },
+          });
+        },
+      });
+      reset();
+    } catch (e) {
+      toast.error(
+        "Car could not be published, you may have already added this car."
+      );
+      const err = e as Error;
+      console.error(err.message);
+    }
+  };
+
+  useEffect(() => {
+    if (brandId) fetchModels({ variables: modelsbyBrandIdVariables(brandId) });
+  }, [brandId]);
+
+  useEffect(() => {
+    if (!mutationReturn) return;
```

Comment:
I think you can have a more descriptive name for this variable, but more important I think you could add the toast to the onCompleted callback from the mutation

Reply:
Huge! I wasn't sure that I could do that, it looks way better and it works nice and neat! Thanks Gerardo!